app/Filament/Pages/AmazonS3.php

```diff
@@ -0,0 +1,296 @@
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Aiding App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    <https://www.canyongbs.com> or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace App\Filament\Pages;
+
+use Throwable;
+use App\Models\User;
+use App\Models\Tenant;
+use Filament\Forms\Form;
+use Filament\Pages\Page;
+use Illuminate\Support\Facades\DB;
+use Filament\Forms\Components\Toggle;
+use Filament\Forms\ComponentContainer;
+use Filament\Forms\Components\Section;
+use Filament\Forms\Contracts\HasForms;
+use Filament\Forms\Components\TextInput;
+use Filament\Notifications\Notification;
+use App\Filament\Clusters\GlobalSettings;
+use Filament\Forms\Concerns\InteractsWithForms;
+use App\Multitenancy\DataTransferObjects\TenantConfig;
+use Filament\Pages\Concerns\CanUseDatabaseTransactions;
+use Filament\Pages\Concerns\HasUnsavedDataChangesAlert;
+use App\Multitenancy\DataTransferObjects\TenantS3FilesystemConfig;
+
+/**
+ * @property ComponentContainer $form
+ */
+class AmazonS3 extends Page implements HasForms
+{
+    use InteractsWithForms;
+    use CanUseDatabaseTransactions;
+    use HasUnsavedDataChangesAlert;
+
+    protected static ?string $navigationIcon = 'heroicon-o-cog-6-tooth';
+
+    protected static ?string $title = 'Amazon S3 Settings';
+
+    protected static ?string $navigationLabel = 'Amazon S3';
+
+    protected static string $view = 'filament.pages.amazon-s3';
+
+    protected static ?string $cluster = GlobalSettings::class;
+
+    protected static ?string $navigationGroup = 'Product Integrations';
+
+    protected static ?int $navigationSort = 100;
+
+    public ?array $data = [];
+
+    public static function canAccess(): bool
+    {
+        /** @var User $user */
+        $user = auth()->user();
+
+        return $user->hasPermissionTo('amazon-s3.manage_s3_settings');
```

Comment:
```suggestion
        return $user->can('amazon-s3.manage_s3_settings');
```
@jayushi-canyon, this was the error as to why it wasn't finding the permission properly. I had used the wrong method before.

Reply:
@Orrison Made the required changes.